simulation_2017/nodes/magic_odomFromGazebo.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+import rospy
+from gazebo_msgs.msg import ModelStates
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import PoseWithCovariance, TwistWithCovariance, PoseArray, PoseStamped
+
+MODELPREFIX = rospy.get_param("magic_modelNamePrefix", "r")
+
+class PFieldNavigator(object):
+
+    def __init__(self):
+        '''
+        Potential field navigator constructor.
+        '''
+        rospy.init_node("magic_gazebo_robot_publisher")
+        self.pub_dict = {}
+        self.pub_dict2 = {}
+        self.pub_dict3 = {}
+        self.last_poses = {}
+        
+        ######################################
+        # Setup ROS publishers
+        #   but nothing automatically publishes
+        ######################################
+        # self.drive_pub = rospy.Publisher("miniBot_poses", PoseArray, queue_size = 10)
+
+        ######################################
+        # Setup ROS subscibers
+        ######################################
+        rospy.Subscriber("/gazebo/model_states", ModelStates, self.gazebo_modelstates_callback)
+
+    def gazebo_modelstates_callback(self, data):
+        '''
+        Callback for model states
+        '''
+        i = 0
+        poses = PoseArray()
+        pose = PoseStamped()
+        while i < len(data.name):
+            if (data.name[i]).startswith(MODELPREFIX):
+                name = data.name[i]
+                # pose = data.pose[i]
```

Comment:
Why is this line commented out?  Now pose is the default PoseStamped message, all values are zero.